src/literal.h

```diff
@@ -489,6 +492,17 @@ template<> struct less<wasm::Literal> {
     WASM_UNREACHABLE("unexpected type");
   }
 };
+inline std::ostream& operator<<(std::ostream& o, wasm::Literals literals) {
```

Comment:
How about putting this alongside the one for singular `Literal`? Looks like that one is a friend in the class, so maybe move it out to here, and maybe keep the bodies in literal.cpp?